### ISSUE TITLE
Compute Engine Exits Abnormally

### ISSUE THREAD

**Issue by griffin28** [visit-dav/live-customer-response#94]
We're trying to [combine `visit-users` email list with GitHub issues](https://github.com/visit-dav/live-customer-response/wiki/How-the-new-GitHub-visit-users-Email-Integration-Works).
When replying on `visit-users`, please reply **only** to emails with a *Subject*
line that includes `[visit-dav/live-customer-response]`.

---

I have compiled Visit 3.1.0 on rhel 7.2 workstation using a simple build. The build executes successfully, except for Sphinx, which I have to interrupt as it cannot find the ‘packages_19.1.tar.gz’ archive and instead tries to go out to the web for it. This workstation has no internet access, so it fails to download the proper archive.
 
My issue arises with the compute engine when I attempt to visualize as result – I am able to read the files in and add the element I want to  visualize however as soon as I select ‘Draw’, I get the following error message:
 
ERROR: The compute engine running on fairfield01.utcaus.com has exited abnormally.
 
WARNING: VisIt could not find a compute engine to use for the plot on host fairfield01.utcaus.com. VisIt will try to launch a compute engine on that host.
 
ERROR: The compute engine running on fairfield01.utcaus.com has exited abnormally.
 
WARNING: VisIt could not find a compute engine to use for the plot on host fairfield01.utcaus.com. VisIt will try to launch a compute engine on that host.
 
I have attempted to run visit with –debug to see if I can understand more, but no additional information prints.
 
Attached is my build log.
 
Thank you,
 
Kevin Mueller

[visit_build_log.tar.gz](https://github.com/visit-dav/live-customer-response/files/4853858/visit_build_log.tar.gz)

**Comment by griffin28** [comment]
Kevin...use `-debug 5` that should generate log files. Please tar.gz them up and attach them to this issue. I will look at those files and see if I can see what's going on.

**Comment by griffin28** [comment]
Hello - Please see my Debug 5 log files attached.

Thank you

Kevin Mueller

[log.tar.gz](https://github.com/visit-dav/live-customer-response/files/4853636/log.tar.gz)

**Comment by biagas** [comment]
Attaching the original build log files that were sent to the users list.
[visit_build_log.tar.gz](https://github.com/visit-dav/live-customer-response/files/4854071/visit_build_log.tar.gz)

**Comment by biagas** [comment]
Thank you for providing the log files.
For your build_visit problem with sphinx, you can add ''--no-sphinx" next time to avoid the issues.

As for the debug log files, I don't see anything in them that indicate the problem. I think your best best is to run it through gdb.
You will need to re-build visit in debug mode. Don't use the build_visit script, just follow the directions below.
It uses the same command to build visit as the build_visit script, with the exception of the build directory name and the CMAKE_BUILD_TYPE being set to Debug instead of release.

```
cd to visit3.1.0
mkdir build-debug
cd build-debug
/local/apps/visit/simple/third_party/cmake/3.9.3/linux-x86_64_gcc-4.8/bin/cmake \
-DVISIT_CONFIG_SITE:FILEPATH=/local/apps/visit/simple/fairfield01.cmake \
-DVISIT_INSTALL_THIRD_PARTY:BOOL=ON \
-DCMAKE_BUILD_TYPE:STRING=Debug \
-DVISIT_C_COMPILER:FILEPATH=gcc \
-DVISIT_CXX_COMPILER:FILEPATH=g++ \
-DVISIT_C_FLAGS:STRING=" -m64 -fPIC -O2" \
-DVISIT_CXX_FLAGS:STRING=" -m64 -fPIC -O2" \
../src
```
Once completed, launch visit (from the build-debug directory) with the gdb debugger, find more info here:
http://visitusers.org/index.php?title=Gdb
especially this section:
http://visitusers.org/index.php?title=Gdb#A_VisIt_component_is_crashing

Hopefully the gdb stack trace will yield more information about where the segv is occurring.

**Comment by griffin28** [comment]
I have finished compiling in debug mode. I ran the debug build using the following command:
 
vglrun visit -debug 5 -gdb engine_ser –xterm
 
And when I attempt to load and visualize my file, the compute engine fails to start and hangs with the ‘compute engine launch progress’ dialog box cycling.
 
Removing the –xterm and running with:
 
vglrun visit -debug 5 -gdb engine_ser
 
Yields the following error: “VisIt could not launch the compute engine on host "fairfield01.utcaus.com".
 
I have attached the log files for both. The ‘xterm.tar.gz’ is the first case and the ‘noxterm.tar.gz’ is the second.
 
Thank you for your help!
 
Kevin

[xterm.tar.gz](https://github.com/visit-dav/live-customer-response/files/4886079/xterm.tar.gz)
[noxterm.tar.gz](https://github.com/visit-dav/live-customer-response/files/4886080/noxterm.tar.gz)
